AirCasting/SessionViews/ThresholdSettings/ThresholdsSettingsViewModel.swift

```diff
@@ -42,10 +42,12 @@ class ThresholdSettingsViewModel: ObservableObject {
             .sorted { $0 < $1 }
         
         // Prevents us from the situation when the user could change thresholds to have the same values
-        newValues = newValues.reversed().mapWithNext { lowerThreshold, higherThreshold in
-            guard lowerThreshold != higherThreshold else { return lowerThreshold - 1 }
-            return lowerThreshold
-        }.sorted { $0 < $1 }
+        while Set(newValues.map { $0 }).count != newValues.count {
```

Comment:
that's nice, how about some more algorithmic and universal approach to this? Take a look at this playground:
```Swift
import UIKit

let hello = [1,1,2,3,4,4,5]

extension Array where Element: Equatable {
    /// Iterates all elements and gets rid of the duplicates using provided strategy
    /// - Parameter strategy: a block invoked when conflict appears. It an `Element` that causes collision as a parameter.
    /// If this block returns `nil`, the new element won't be added to the resulting array.
    /// If it returns a value, this value will be added to the resulting array.
    /// - Returns: A new `Array` with conflicts resolved.
    func byRemovingDuplicates(strategy: (_ colliding: Element) -> Element?) -> Self {
        var buffer: Self = []
        for element in self {
            print("Element: \(element)")
            appendBuffer(&buffer, with: element, strategy: strategy)
        }
        return buffer
    }
    
    private func appendBuffer(_ buffer: inout Self, with value: Element, strategy: (_ colliding: Element) -> Element?) {
        guard buffer.contains(value) else {
            print("Appending buffer with \(value)")
            buffer.append(value)
            return
        }
        
        if let modified = strategy(value) {
            print("\(value) already present, resolving to \(modified)...")
            appendBuffer(&buffer, with: modified, strategy: strategy)
        }
    }
}

extension Array where Element: Equatable & Numeric {
    var incrementRemovalStrategy: (Element) -> Element? {
        { $0.increment() }
    }
}

extension Numeric {
    func increment() -> Self {
        self + 1
    }
}

print(hello.byRemovingDuplicates(strategy: hello.incrementRemovalStrategy))
```

What do you think?

Reply:
I changed small things there, please have a look what do you think?